syntax = "proto2";

package nqueens;

enum MasterMsgID {
    MASTER_MSG_CALL = 1;
    MASTER_MSG_ASSIGN_WORK = 2;
};

enum SlaveMsgID {
    SLAVE_MSG_ASK_FOR_WORK = 1;
    SLAVE_MSG_TASK_RESULT = 2;
};

enum ErrorCode {
    ERROR_NO_ERROR = 0;
   
    ERROR_MASTER_NO_MORE_WORK = 1;
};

message MasterCall {    // MSGID : MASTER_MSG_CALL
    required string password = 1;
}

message MasterAssignWork {    // MSGID : MASTER_MSG_ASSIGN_WORK
    required int32 error_code = 1;
    optional int32 n = 2;
    repeated int32 constraint = 3;
}

message SlaveAskForWork {    // MSGID : SLAVE_MSG_ASK_FOR_WORK
    required string password = 1;
}

message SlaveTaskResult {    // MSGID : SLAVE_MSG_TASK_RESULT
    required int64 result = 1;
}
